### PR TITLE
fix: mise use -p . panics

### LINKED ISSUE #3033: `mise use -p .` panics
**Describe the bug**
Whenever using `mise use -p . <tool-here>`, mise panics.

The commit that seems to have introduced this (as found through a `git bisect`) is b36acc3288d5b087ef126593037d7c10232635ab

**To Reproduce**

```
$ mise use -p . java@temurin-21                                                                                                                  at 20:48:53
The application panicked (crashed).
Message:  called `Option::unwrap()` on a `None` value
Location: src/config/config_file/mod.rs:329

Backtrace omitted. Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.
```

**Expected behavior**
I'd expect mise to find or create a new config file in the PWD.

**`mise doctor` output**

```text
version: 2024.11.12 macos-arm64 (2024-11-14)
activated: yes
shims_on_path: no

build_info:
  Target: aarch64-apple-darwin
  Features: DEFAULT, NATIVE_TLS
  Built: Thu, 14 Nov 2024 08:03:00 +0000
  Rust Version: rustc 1.82.0 (f6e511eec 2024-10-15) (Homebrew)
  Profile: release

shell:
  /opt/homebrew/bin/fish
  fish, version 3.7.1

dirs:
  data: ~/.local/share/mise
  config: ~/.config/mise
  cache: ~/Library/Caches/mise
  state: ~/.local/state/mise
  shims: ~/.local/share/mise/shims

config_files:
  ~/.config/mise/config.toml
  ~/Projects/.mise.toml

backends:
  aqua
  asdf
  cargo
  core
  go
  npm
  pipx
  spm
  ubi
  vfox
  unknown

plugins:
  maven    https://github.com/halcyon/asdf-maven.git#83fb4db
  pnpm     https://github.com/jonathanmorley/asdf-pnpm.git#305baff
  pulumi   https://github.com/canha/asdf-pulumi.git#99d675b
  quarkus  https://github.com/asdf-community/asdf-quarkus.git#d00eb32
  ruff     https://github.com/simhem/asdf-ruff#97b4f67
  usage    https://github.com/jdx/mise-usage.git#fe3888a
  uv       https://github.com/asdf-community/asdf-uv.git#54b1c75
  yarn     https://github.com/mise-plugins/asdf-yarn.git#74ea3b9

toolset:
  asdf:mise-plugins/asdf-maven@3.9.9
  core:java@temurin-17.0.13+11
  asdf:asdf-community/asdf-quarkus@3.16.1
  core:python@3.13.0
  asdf:jonathanmorley/asdf-pnpm@9.12.3
  ubi:jdx/usage@1.0.1
  core:bun@1.1.33
  asdf:asdf-community/asdf-uv@0.4.29

env_vars:
  MISE_SHELL=fish

settings:
  activate_aggressive = false
  all_compile = false
  always_keep_download = false
  always_keep_install = false
  asdf_compat = false
  cache_prune_age = "30d"
  ci = false
  color = true
  debug = false
  disable_backends = []
  disable_default_registry = false
  disable_hints = []
  disable_tools = []
  experimental = false
  fetch_remote_versions_cache = "1h"
  fetch_remote_versions_timeout = "10s"
  go_default_packages_file = "~/.default-go-packages"
  go_download_mirror = "https://dl.google.com/go"
  go_repo = "https://github.com/golang/go"
  go_set_gopath = false
  go_set_goroot = true
  go_skip_checksum = false
  http_timeout = "30s"
  jobs = 4
  legacy_version_file = true
  legacy_version_file_disable_tools = []
  libgit2 = true
  lockfile = false
  log_level = "info"
  not_found_auto_install = true
  paranoid = false
  pin = false
  plugin_autoupdate_last_check_duration = "7d"
  quiet = false
  raw = false
  task_timings = false
  trace = false
  trusted_config_paths = []
  unix_default_file_shell_args = ["sh"]
  unix_default_inline_shell_args = [
      "sh",
      "-c",
  ]
  use_file_shell_for_executable_tasks = false
  use_versions_host = true
  verbose = false
  windows_default_file_shell_args = [
      "cmd",
      "/c",
  ]
  windows_default_inline_shell_args = [
      "cmd",
      "/c",
  ]
  windows_executable_extensions = [
      "exe",
      "bat",
      "cmd",
      "com",
      "ps1",
      "vbs",
  ]
  yes = false

  [cargo]
  binstall = true

  [node]

  [npm]
  bun = false

  [pipx]
  uvx = false

  [python]
  default_packages_file = "~/.default-python-packages"
  pyenv_repo = "https://github.com/pyenv/pyenv.git"
  venv_auto_create = false
  venv_stdlib = false

  [ruby]
  default_packages_file = "~/.default-gems"
  ruby_build_repo = "https://github.com/rbenv/ruby-build.git"
  ruby_install = false
  ruby_install_repo = "https://github.com/postmodern/ruby-install.git"

  [status]
  missing_tools = "if_other_versions_installed"
  show_env = false
  show_tools = false
No warnings found
1 problem found:

1. new mise version 2024.11.13 available, currently on 2024.11.12
```

### PR DESCRIPTION
**Pull request by roele**
Fixes #3033

### CODE REVIEW

**Comment by jdx** [comment]
test failed

File: e2e/cli/test_use [hunk 1]
```diff
@@ -35,6 +35,10 @@ dummy = "1.0.0"'
 mise use --rm dummy --path mise.local.toml
 assert "cat mise.local.toml" ""
 
+mise use dummy@1 --path .
```
Comment (jdx) [inline e2e/cli/test_use:38]: this makes the test pass but now it's not testing a valid use-case. I should be able to run `msie use dummy@1 --path .` if no file exists there and mise should create one

Comment (jdx) [reply e2e/cli/test_use:38]: we use [path-absolutize](https://crates.io/crates/path-absolutize) which should work for this

Comment (roele) [reply e2e/cli/test_use:38]: What makes the test pass is the fallback i added with `let abs_from_dir = from_dir.canonicalize().unwrap_or_else(|_| from_dir.clone());`. This is supposed to be an additional test that should test the initial reported issue.

Comment (roele) [reply e2e/cli/test_use:38]: I see `absolutize` is a better approach than `canonicalize`, will replace it.

Comment (jdx) [reply e2e/cli/test_use:38]: oh sorry, I should've noticed this was only adding a test case, still yeah, absolutize will make this a bit cleaner

**Pull request merged**